diff --git a/perf/trace-events.h b/perf/trace-events.h
index 535ca327..c3d4cfd0 100644
--- a/perf/trace-events.h
+++ b/perf/trace-events.h
@@ -45,6 +45,7 @@ enum {
 	TRACE_EVENT_LOCK_CLEAR = 0x30,
 	TRACE_EVENT_THREAD_PRIORITY = 0x31,
 	TRACE_EVENT_PROCESS_KILL = 0x32,
+	TRACE_EVENT_MSG_SEND = 0x33,
 };
 
 
@@ -71,8 +72,14 @@ void _trace_updateLockEpoch(lock_t *lock);
  * NOTE: The ev structure passed to PERF_{META,EVENT}_BODY must match the
  * field struct declared in the tsdl/metadata for a given event_id.
  */
+#define NO_EVENT(ev, ts) \
+	do { \
+		(void)ev; \
+		(void)ts; \
+	} while (0)
 #define TRACE_META_BODY(event_id, ev, ts, ...)  TRACE_EVENT_BODY_CHAN((u8)(trace_channel_meta), (event_id), (ev), (ts), __VA_ARGS__)
-#define TRACE_EVENT_BODY(event_id, ev, ts, ...) TRACE_EVENT_BODY_CHAN((u8)(trace_channel_event), (event_id), (ev), (ts), __VA_ARGS__)
+#define TRACE_EVENT_BODY(event_id, ev, ts, ...) NO_EVENT((ev), (ts))
+#define TRACE_MSG_BODY(event_id, ev, ts, ...)   TRACE_EVENT_BODY_CHAN((u8)(trace_channel_event), (event_id), (ev), (ts), __VA_ARGS__)
 
 
 /* assumes lock->spinlock is set */
@@ -305,4 +312,24 @@ static inline void trace_eventProcessKill(const process_t *p)
 }
 
 
+static inline void trace_eventMsgSend(int tid, u32 port, u8 type, size_t isize, size_t osize)
+{
+	struct {
+		u16 tid;
+		u16 port;
+		u8 type;
+		u32 isize;
+		u32 osize;
+	} __attribute__((packed)) ev;
+
+	TRACE_MSG_BODY(TRACE_EVENT_MSG_SEND, ev, NULL, {
+		ev.tid = (u16)tid;
+		ev.port = (u16)port;
+		ev.type = type;
+		ev.isize = (u32)isize;
+		ev.osize = (u32)osize;
+	});
+}
+
+
 #endif
diff --git a/perf/tsdl/metadata b/perf/tsdl/metadata
index 9a5be86b..5f4abc5b 100644
--- a/perf/tsdl/metadata
+++ b/perf/tsdl/metadata
@@ -194,3 +194,15 @@ event {
 		u16 pid;
 	};
 };
+
+event {
+	name = msg_send;
+	id = 0x33;
+	fields := struct {
+		u16 tid;
+		u16 port;
+		u8 type;
+		u32 isize;
+		u32 osize;
+	};
+};
diff --git a/proc/msg.c b/proc/msg.c
index 65a826d0..1bfb77ee 100644
--- a/proc/msg.c
+++ b/proc/msg.c
@@ -15,6 +15,7 @@
 
 #include "include/errno.h"
 #include "lib/lib.h"
+#include "perf/trace-events.h"
 #include "proc.h"
 
 
@@ -356,6 +357,8 @@ int proc_send(u32 port, msg_t *msg)
 	spinlock_ctx_t sc;
 	int state;
 
+	trace_eventMsgSend(proc_getTid(proc_current()), port, msg->type, msg->i.size, msg->o.size);
+
 	/* TODO - check if msg pointer belongs to user vm_map */
 	if (msg == NULL) {
 		return -EINVAL;
