diff -Naur binutils-2.34/bfd/config.bfd binutils-2.34-phoenix/bfd/config.bfd
--- binutils-2.34/bfd/config.bfd	2020-01-18 14:55:47.000000000 +0100
+++ binutils-2.34-phoenix/bfd/config.bfd	2020-03-18 14:08:26.037301292 +0100
@@ -758,6 +758,10 @@
     targ_defvec=i386_elf32_vxworks_vec
     targ_underscore=yes
     ;;
+  i[3-7]86-*-phoenix*)
+    targ_defvec=i386_elf32_vec
+    targ_selfvecs=
+    ;;
 
   ia16-*-elf)
     targ_defvec=i386_elf32_vec
diff -Naur binutils-2.34/gas/configure.tgt binutils-2.34-phoenix/gas/configure.tgt
--- binutils-2.34/gas/configure.tgt	2020-01-18 14:55:47.000000000 +0100
+++ binutils-2.34-phoenix/gas/configure.tgt	2020-03-18 14:09:53.368838497 +0100
@@ -259,6 +259,7 @@
   i386-*-*nt*)				fmt=coff em=pe ;;
   i386-*-rdos*)				fmt=elf ;;
   i386-*-darwin*)			fmt=macho ;;
+  i386-*-phoenix*)			fmt=elf ;;
 
   ia16-*-elf*)				fmt=elf ;;
 
diff -Naur binutils-2.34/ld/configure.tgt binutils-2.34-phoenix/ld/configure.tgt
--- binutils-2.34/ld/configure.tgt	2020-01-18 14:55:48.000000000 +0100
+++ binutils-2.34-phoenix/ld/configure.tgt	2020-03-18 14:11:13.700432001 +0100
@@ -402,6 +402,9 @@
 			;;
 i[3-7]86-*-vxworks*)	targ_emul=elf_i386_vxworks
 			;;
+i[3-7]86-*-phoenix*)	targ_emul=elf_i386_phoenix
+			targ_extra_emuls=elf_i386
+			;;
 i[3-7]86-*-chaos)	targ_emul=elf_i386_chaos
 			;;
 i[3-7]86-*-nacl*)	targ_emul=elf_i386_nacl
diff -Naur binutils-2.34/ld/emulparams/elf_i386_phoenix.sh binutils-2.34-phoenix/ld/emulparams/elf_i386_phoenix.sh
--- binutils-2.34/ld/emulparams/elf_i386_phoenix.sh	1970-01-01 01:00:00.000000000 +0100
+++ binutils-2.34-phoenix/ld/emulparams/elf_i386_phoenix.sh	2020-03-18 14:12:39.096016999 +0100
@@ -0,0 +1 @@
+. ${srcdir}/emulparams/elf_i386.sh
diff -Naur binutils-2.34/ld/Makefile.am binutils-2.34-phoenix/ld/Makefile.am
--- binutils-2.34/ld/Makefile.am	2020-01-30 08:15:56.000000000 +0100
+++ binutils-2.34-phoenix/ld/Makefile.am	2020-03-18 14:15:37.443195906 +0100
@@ -295,6 +295,7 @@
 	eelf_i386_nacl.c \
 	eelf_i386_sol2.c \
 	eelf_i386_vxworks.c \
+	eelf_i386_phoenix.c \
 	eelf_iamcu.c \
 	eelf_s390.c \
 	eh8300elf.c \
@@ -783,6 +784,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_nacl.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_sol2.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_vxworks.Pc@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_phoenix.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_iamcu.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_s390.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eh8300elf.Pc@am__quote@
diff -Naur binutils-2.34/ld/Makefile.in binutils-2.34-phoenix/ld/Makefile.in
--- binutils-2.34/ld/Makefile.in	2020-02-01 12:50:25.000000000 +0100
+++ binutils-2.34-phoenix/ld/Makefile.in	2020-03-18 14:16:31.746955845 +0100
@@ -785,6 +785,7 @@
 	eelf_i386_nacl.c \
 	eelf_i386_sol2.c \
 	eelf_i386_vxworks.c \
+	eelf_i386_phoenix.c \
 	eelf_iamcu.c \
 	eelf_s390.c \
 	eh8300elf.c \
@@ -1391,6 +1392,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_nacl.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_sol2.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_vxworks.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_phoenix.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_iamcu.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_k1om.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_k1om_fbsd.Po@am__quote@
@@ -2387,6 +2389,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_nacl.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_sol2.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_vxworks.Pc@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_phoenix.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_iamcu.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_s390.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eh8300elf.Pc@am__quote@
