--- gcc-9.3.0-orig/gcc/config.gcc	2020-03-12 12:07:21.000000000 +0100
+++ gcc-9.3.0/gcc/config.gcc	2020-03-18 19:25:57.937836171 +0100
@@ -1205,7 +1205,9 @@
 	tm_file="elfos.h arm/unknown-elf.h arm/elf.h arm/bpabi.h"
 	tm_file="${tm_file} newlib-stdint.h phoenix.h"
 	tm_file="${tm_file} arm/aout.h arm/arm.h"
-	tmake_file="${tmake_file} arm/t-arm arm/t-bpabi arm/t-phoenix"
+	tmake_file="${tmake_file} arm/t-arm arm/t-bpabi arm/t-aprofile arm/t-rmprofile arm/t-multilib"
+	with_float="soft" # "auto" fpu
+	TM_MULTILIB_CONFIG="aprofile,rmprofile"
 	target_cpu_cname="arm7tdmi"
 	;;
 arm*-*-eabi* | arm*-*-symbianelf* | arm*-*-rtems* | arm*-*-fuchsia*)
